src/matching/basin_enclosure.jl

```diff
@@ -54,6 +54,25 @@ Basin enclosure is a concept similar to "basin instability" in [Ritchie2023](@ci
     consecutive_lost_steps::Int = 1000
 end
 
+function next_key(dict)
+    allkeys = collect(values(dict))
+    maxkey = maximum(allkeys)
+    return maxkey+1
+end
```

Comment:
You have to use the existing infrastructure with `next_id`. It is already passed to the `matching_map` function. Please see how `next_id` is handled in the `matching_map` implementation for `MatchBySSSetDistance`.